biorxiv/article_distances/biorxiv_corrected_published_rate.py

```diff
@@ -0,0 +1,520 @@
+        "doi":"last",
+        "published_doi":"first",  
+        "pmcid":"first", 
+        "pmcoa":"first"
+    })
+    .reset_index(drop=True)
+    .assign(published=lambda x: x.published_doi.apply(lambda y: "/" in str(y)))
+)
+print(biorxiv_journal_df.shape)
+biorxiv_journal_df.head()
+
+
+# In[5]:
+
+
+print(f"Total Published: {biorxiv_journal_df.published.sum()}")
+print(f"Total Published in PMCOA: {biorxiv_journal_df.pmcoa.sum()}")
+
+
+# In[6]:
+
+
+annotated_links_df =(
+    pd.read_csv(
+        Path("output/annotated_links")/
+        Path("potential_biorxiv_pmc_links.tsv"), 
+        sep="\t"
+    )
+    .assign(published=True)
+    .rename(index=str, columns={"biorxiv_doi":"doi"})
+    [["document", "doi", "published", "distance_bin"]]
+)
+print(annotated_links_df.shape)
+annotated_links_df.head()
+
+
+# In[7]:
+
+
+missing_link_mapper = {
+    row['doi']:[
+        row['published'],
+        True
+    ]
+    for row in (
+        annotated_links_df
+        .query("distance_bin in ['[0, 25%ile)', '[25%ile, 50%ile)']")
+        [["doi", "published"]]
+        .to_dict(orient="records")
+    )
+}
+new_link_count = len(missing_link_mapper)
+print(f"Novel links filled: {new_link_count}")
+
+
+# In[8]:
+
+
+remaining_links = {
+    row['doi']:[
+        row['published'],
+        row['pmcoa']
+    ]
+    for row in (
+        biorxiv_journal_df
+        [["doi", "published", "pmcoa"]]
+        .to_dict(orient="records")
+    ) 
+    if row['doi'] not in missing_link_mapper
+}
+print(len(remaining_links))
+
+
+# In[9]:
+
+
+missing_link_mapper.update(remaining_links)
+print(len(missing_link_mapper))
+
+
+# In[10]:
+
+
+updated_biorxiv_journal_df = (
+    pd.DataFrame
+    .from_records([
+        {
+            "doi":row[0], 
+            "published":row[1][0], 
+            "pmcoa":row[1][1]
+        } 
+        for row in list(remaining_links.items())
+    ])
+    .merge(
+        biorxiv_journal_df
+         .drop(["published", "pmcoa"], axis=1), 
+        on="doi"
+    )
+)
+print(updated_biorxiv_journal_df.shape)
+updated_biorxiv_journal_df.head()
+
+
+# ## Gather date published for each document
+
+# In[11]:
+
+
+file_path = Path("output/biorxiv_articles_published_date.tsv")
+
+
+# In[12]:
+
+
+if not file_path.exists():
+    data_records = []
+    xml_parser = ET.XMLParser(encoding='UTF-8', recover=True)
+    for document in tqdm(updated_biorxiv_journal_df.document.tolist()):
+        root = (
+            ET
+            .parse(
+                f"../biorxiv_articles/{document}", 
+                parser=xml_parser
+            )
+            .getroot()
+        )
+
+        date_node = root.xpath(
+            "//front/article-meta/history/date"
+        )
+
+        if len(date_node) == 0:
+            print("Error in xml:")
+            print(document)
+            continue
+
+        elif len(date_node) < 2:
+            date_node = date_node[0]
+
+        else:
+            date_node = date_node[1]
```

Comment:
What do these cases correspond to? Does biorxiv have two different date formats in the metadata?

Reply:
> What do these cases correspond to? Does biorxiv have two different date formats in the metadata?

BioRixv has a submitted date and accepted date. I'm using the date the preprints were submitted for my analysis. As with all things though there are errors within the bioRxiv xml files and I needed to account for that.